diff --git a/FDS_Source/main.f90 b/FDS_Source/main.f90
index e045187e4..95b123576 100644
--- a/FDS_Source/main.f90
+++ b/FDS_Source/main.f90
@@ -97,7 +97,7 @@ WALL_CLOCK_START = WALL_CLOCK_TIME()
  
 ! Assign a compilation date (All Nodes)
 
-WRITE(VERSION_STRING,'(A)') '5.5.3'
+WRITE(VERSION_STRING,'(A)') '5.5.3+gbf0a6a88f'
 
 IF (INDEX(mainrev,':',BACK=.TRUE.)>0) THEN
    WRITE(REVISION_DATE,'(A)',IOSTAT=IOS,ERR=5) mainrev(INDEX(mainrev,':')+1:LEN_TRIM(mainrev)-2)
