diff --git a/FDS_Compilation/makefile b/FDS_Compilation/makefile
index 8c7116292..892527547 100644
--- a/FDS_Compilation/makefile
+++ b/FDS_Compilation/makefile
@@ -19,14 +19,14 @@ ifeq ($(shell echo "check_quotes"),"check_quotes")
 else
 # linux/osx
   GIT_HASH := $(shell git describe --long --dirty)
-  GIT_DATE := $(shell git log -1 --format=%cd)
-  BUILD_DATE := $(shell date "+%b %d, %Y  %T")
+  GIT_DATE := $(shell date "+%b %d, %Y  %T" --date='@${SOURCE_DATE_EPOCH}')
+  BUILD_DATE := $(shell date "+%b %d, %Y  %T" --date='@${SOURCE_DATE_EPOCH}')
 # The preprocessor used by xlf uses comma as separation between preprocessing commands.
 # Commas needs to be escaped if they are to be used as part of a value in a preprocessing variable.
-  BUILD_DATE_XLF := $(shell date "+%b %d\, %Y  %T")  
+  BUILD_DATE_XLF := $(shell date "+%b %d\, %Y  %T")
 endif
 
-GITINFO_BASE=-DGITHASH_PP=\"$(GIT_HASH)\" -DGITDATE_PP=\""$(GIT_DATE)\""
+GITINFO_BASE=-DGITHASH_PP=\"$(build_version)+g$(commit)\" -DGITDATE_PP=\""$(GIT_DATE)\""
 GITINFO=-fpp $(GITINFO_BASE) -DBUILDDATE_PP=\""$(BUILD_DATE)\""
 GITINFOGNU=-cpp $(GITINFO_BASE) -DBUILDDATE_PP=\""$(BUILD_DATE)\""
 GITINFOXLF=-qsuffix=cpp=f90 '-WF,-DGITHASH_PP=\"$(GIT_HASH)\",-DGITDATE_PP=\"$(GIT_DATE)\",-DBUILDDATE_PP=\"$(BUILD_DATE_XLF)\"'
