diff --git a/CMakeLists.txt b/CMakeLists.txt
index 92364896ec..88789e6e57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,7 +29,6 @@ add_executable(fds
     Source/hvac.f90
     Source/ieva.f90
     Source/init.f90
-    Source/irad.f90
     Source/main.f90
     Source/mass.f90
     Source/mesh.f90
@@ -38,6 +37,7 @@ add_executable(fds
     Source/prec.f90
     Source/pres.f90
     Source/radi.f90
+    Source/rcal.f90
     Source/read.f90
     Source/samr.f90
     Source/scrc.f90
