diff --git a/CMakeLists.txt b/CMakeLists.txt
index abcde8b87..593952f35 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.24 FATAL_ERROR)
 
 project(
     fds
-    VERSION 6.1.2
+    VERSION 6.2.0
     LANGUAGES Fortran
 )
 enable_language(Fortran)
@@ -42,6 +42,7 @@ add_executable(fds
     FDS_Source/read.f90
     FDS_Source/samr.f90
     FDS_Source/scrc.f90
+    FDS_Source/soot.f90
     FDS_Source/smvv.f90
     FDS_Source/turb.f90
     FDS_Source/type.f90
