diff --git a/Source/main.f90 b/Source/main.f90
index 8ef555de4..d8aef6afb 100644
--- a/Source/main.f90
+++ b/Source/main.f90
@@ -1302,7 +1302,6 @@ CALL DUMP_CPU_TIME
 
 IF (USE_MPI) THEN
    IF (VERBOSE) WRITE(LU_ERR,'(A,I6,A)') 'MPI process ',MYID,' has completed'
-   CALL MPI_FINALIZE(IERR)
 ENDIF
 
 IF (MYID==0) THEN
@@ -1342,6 +1341,10 @@ IF (MYID==0) THEN
 
 ENDIF
 
+IF (USE_MPI) THEN
+   CALL MPI_FINALIZE(IERR)
+ENDIF
+
 STOP
 
 END SUBROUTINE END_FDS
diff --git a/Source/smvv.f90 b/Source/smvv.f90
index cb636a50f..b9319cae0 100644
--- a/Source/smvv.f90
+++ b/Source/smvv.f90
@@ -800,7 +800,7 @@ SUBROUTINE SMOKE3D_TO_FILE(LU_SMOKE3D,LU_SMOKE3D_SIZE,TIME,DX,EXTCOEF,SMOKE_TYPE
   REAL(FB), INTENT(IN) :: HRRPUV_MAX_SMV
 
   INTEGER, PARAMETER :: SOOT=1, FIRE=2, OTHER=3
-  CHARACTER(LEN=1), DIMENSION(:), POINTER :: BUFFER_IN, BUFFER_OUT
+  CHARACTER(LEN=1), DIMENSION(:), ALLOCATABLE :: BUFFER_IN, BUFFER_OUT
   INTEGER :: NCHARS_IN
   REAL(FB) :: FACTOR, CUTMAX
   REAL(FB) :: VAL_FDS,VAL_SMV,MAXVAL
@@ -869,7 +869,7 @@ SUBROUTINE RLE(BUFFER_IN, NCHARS_IN, BUFFER_OUT, NCHARS_OUT)
 
   INTEGER, INTENT(IN) :: NCHARS_IN
   CHARACTER(LEN=1), INTENT(IN), DIMENSION(NCHARS_IN) :: BUFFER_IN
-  CHARACTER(LEN=1), INTENT(OUT), DIMENSION(:), POINTER :: BUFFER_OUT
+  CHARACTER(LEN=1), INTENT(OUT), DIMENSION(:) :: BUFFER_OUT
   INTEGER, INTENT(OUT) :: NCHARS_OUT
 
   CHARACTER(LEN=1) :: MARK=CHAR(255),THISCHAR,LASTCHAR
diff --git a/Build/makefile b/Build/makefile
index 5bda0941f..29a1c2dae 100644
--- a/Build/makefile
+++ b/Build/makefile
@@ -75,7 +86,7 @@ else
 ifneq ("$(wildcard ${MKLROOT}/lib/intel64/libmkl_blacs_intelmpi_lp64.a)","")
 # Use cluster_sparse_solver with intelmpi blacs
   FFLAGSMKL_INTEL = -DWITH_MKL -I$(MKLROOT)/include
-  LFLAGSMKL_INTEL = -Wl,--start-group ${MKLROOT}/lib/intel64/libmkl_intel_lp64.a ${MKLROOT}/lib/intel64/libmkl_core.a ${MKLROOT}/lib/intel64/libmkl_intel_thread.a ${MKLROOT}/lib/intel64/libmkl_blacs_intelmpi_lp64.a -Wl,--end-group -lpthread -lm -ldl
+  LFLAGSMKL_INTEL = -Wl,--start-group ${MKLROOT}/lib/intel64/libmkl_intel_lp64.a ${MKLROOT}/lib/intel64/libmkl_core.a ${MKLROOT}/lib/intel64/libmkl_intel_thread.a ${MKLROOT}/lib/intel64/libmkl_blacs_intelmpi_lp64.a -Wl,--end-group -lpthread -ldl
 endif
 ifneq ("$(wildcard ${MKLROOT}/lib/intel64/libmkl_blacs_openmpi_lp64.a)","")
 # Use cluster_sparse_solver with openmpi blacs
