diff --git a/Source/main.f90 b/Source/main.f90
index b9851c63c..c14f06020 100644
--- a/Source/main.f90
+++ b/Source/main.f90
@@ -1350,7 +1350,6 @@ LOGICAL :: OPN
 IF (STOP_STATUS==NO_STOP .OR. STOP_STATUS==USER_STOP) CALL DUMP_TIMERS
 
 IF (VERBOSE) WRITE(LU_ERR,'(A,I6,A)') 'MPI process ',MYID,' has completed'
-CALL MPI_FINALIZE(IERR)
 
 IF (MYID==0) THEN
 
@@ -1389,6 +1388,8 @@ IF (MYID==0) THEN
 
 ENDIF
 
+CALL MPI_FINALIZE(IERR)
+
 STOP
 
 END SUBROUTINE END_FDS
diff --git a/Build/makefile b/Build/makefile
index 98ae0a3ba..79b1fb3ec 100644
--- a/Build/makefile
+++ b/Build/makefile
@@ -75,7 +75,7 @@ else
 ifneq ("$(wildcard ${MKLROOT}/lib/intel64/libmkl_blacs_intelmpi_lp64.a)","")
 # Use cluster_sparse_solver with intelmpi blacs
   FFLAGSMKL_INTEL = -DWITH_CLUSTER_SPARSE_SOLVER -I$(MKLROOT)/include
-  LFLAGSMKL_INTEL = -Wl,--start-group ${MKLROOT}/lib/intel64/libmkl_intel_lp64.a ${MKLROOT}/lib/intel64/libmkl_core.a ${MKLROOT}/lib/intel64/libmkl_intel_thread.a ${MKLROOT}/lib/intel64/libmkl_blacs_intelmpi_lp64.a -Wl,--end-group -lpthread -lm -ldl
+  LFLAGSMKL_INTEL = -Wl,--start-group ${MKLROOT}/lib/intel64/libmkl_intel_lp64.a ${MKLROOT}/lib/intel64/libmkl_core.a ${MKLROOT}/lib/intel64/libmkl_intel_thread.a ${MKLROOT}/lib/intel64/libmkl_blacs_intelmpi_lp64.a -Wl,--end-group -lpthread -ldl
 endif
 ifneq ("$(wildcard ${MKLROOT}/lib/intel64/libmkl_blacs_openmpi_lp64.a)","")
 # Use cluster_sparse_solver with openmpi blacs
