diff --git a/Build/impi_intel_linux_64/make_fds.sh b/Build/impi_intel_linux_64/make_fds.sh
index 2317834e8..5e704f1c1 100755
--- a/Build/impi_intel_linux_64/make_fds.sh
+++ b/Build/impi_intel_linux_64/make_fds.sh
@@ -4,4 +4,4 @@ dir=`pwd`
 target=${dir##*/}
 
 echo Building $target with Intel MPI
-make -j4 VPATH="../../Source" -f ../makefile $target
+make -j4 VPATH="../../Source" INTEL_IFORT=ifx INTEL_MPI_IFORT=mpiifx -f ../makefile $target
diff --git a/Build/makefile b/Build/makefile
index d75026b56..0792daf33 100644
--- a/Build/makefile
+++ b/Build/makefile
@@ -11,6 +11,17 @@
 
 VPATH = ../Source
 
+I_IFORT=ifort
+MPI_I_IFORT=mpifort
+
+ifdef INTEL_IFORT
+  I_IFORT=$(INTEL_IFORT)
+endif
+
+ifdef INTEL_MPI_IFORT
+  MPI_I_IFORT=$(INTEL_MPI_IFORT)
+endif
+
 ifeq ($(shell echo "check_quotes"),"check_quotes")
 # windows
   GIT_HASH := $(shell ..\..\Utilities\Scripts\githash)
@@ -150,7 +161,7 @@ impi_intel_win_64 : MPILIB = "$(I_MPI_RELEASE_ROOT)\impi.lib"
 impi_intel_win_64 : MPIINCLUDE = "$(I_MPI_ROOT)\intel64\include"
 impi_intel_win_64 : FFLAGS = /Qipo /O2 /I$(MPIINCLUDE) /traceback /wrap-margin- $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_win_64 : FOPENMPFLAGS = /Qopenmp
-impi_intel_win_64 : FCOMPL = ifort
+impi_intel_win_64 : FCOMPL = $(I_IFORT)
 impi_intel_win_64 : obj = fds_impi_win_64
 impi_intel_win_64 : setup_win $(objwin_mpi)
 	$(FCOMPL) -o $(obj) $(FFLAGS) $(FOPENMPFLAGS) /F1000000000 $(objwin_mpi) $(MPILIB) $(LFLAGSMKL_INTEL)
@@ -159,7 +170,7 @@ impi_intel_win_64_dv : MPILIB = "$(I_MPI_RELEASE_ROOT)\impi.lib"
 impi_intel_win_64_dv : MPIINCLUDE = "$(I_MPI_ROOT)\intel64\include"
 impi_intel_win_64_dv : FFLAGS = /O1 /I$(MPIINCLUDE) /traceback /wrap-margin- $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_win_64_dv : FOPENMPFLAGS = /Qopenmp
-impi_intel_win_64_dv : FCOMPL = ifort
+impi_intel_win_64_dv : FCOMPL = $(I_IFORT)
 impi_intel_win_64_dv : obj = fds_impi_win_64_dv
 impi_intel_win_64_dv : setup_win $(objwin_mpi)
 	$(FCOMPL) -o $(obj) $(FFLAGS) $(FOPENMPFLAGS) /F1000000000 $(objwin_mpi) $(MPILIB) $(LFLAGSMKL_INTEL)
@@ -168,7 +179,7 @@ impi_intel_win_64_db : MPILIB = "$(I_MPI_DEBUG_ROOT)\impi.lib"
 impi_intel_win_64_db : MPIINCLUDE = "$(I_MPI_ROOT)\intel64\include"
 impi_intel_win_64_db : FFLAGS = /Od /nologo /debug:all /I$(MPIINCLUDE) /Z7 /extend_source:132 /warn:unused /warn:nointerfaces /Qtrapuv /fp:strict /fp:except /traceback /check:all /stand:f08 /wrap-margin- $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_win_64_db : FOPENMPFLAGS = /Qopenmp
-impi_intel_win_64_db : FCOMPL = ifort
+impi_intel_win_64_db : FCOMPL = $(I_IFORT)
 impi_intel_win_64_db : obj = fds_impi_win_64_db
 impi_intel_win_64_db : setup_win $(objwin_mpi)
 	$(FCOMPL) -o $(obj) $(FFLAGS) $(FOPENMPFLAGS) /F1000000000 $(objwin_mpi) $(MPILIB) $(LFLAGSMKL_INTEL)
@@ -177,7 +188,7 @@ impi_intel_win_64_db : setup_win $(objwin_mpi)
 impi_intel_linux_64 : FFLAGS = -m64 -O2 -ipo -mt_mpi -traceback -no-wrap-margin $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_linux_64 : LFLAGSMKL = $(LFLAGSMKL_INTEL)
 impi_intel_linux_64 : LFLAGS = -static-intel
-impi_intel_linux_64 : FCOMPL = mpiifort
+impi_intel_linux_64 : FCOMPL = $(MPI_I_IFORT)
 impi_intel_linux_64 : FOPENMPFLAGS = -qopenmp -qopenmp-link static -liomp5
 impi_intel_linux_64 : obj = fds_impi_intel_linux_64
 impi_intel_linux_64 : setup $(obj_mpi)
@@ -186,7 +197,7 @@ impi_intel_linux_64 : setup $(obj_mpi)
 impi_intel_linux_64_db : FFLAGS = -m64 -mt_mpi -check -warn all -O0 -auto -WB -traceback -g -fpe0 -fltconsistency -stand:f08 -no-wrap-margin $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_linux_64_db : LFLAGSMKL = $(LFLAGSMKL_INTEL)
 impi_intel_linux_64_db : LFLAGS = -static-intel
-impi_intel_linux_64_db : FCOMPL = mpiifort
+impi_intel_linux_64_db : FCOMPL = $(MPI_I_IFORT)
 impi_intel_linux_64_db : FOPENMPFLAGS = -qopenmp -qopenmp-link static -liomp5
 impi_intel_linux_64_db : obj = fds_impi_intel_linux_64_db
 impi_intel_linux_64_db : setup $(obj_mpi)
@@ -195,7 +206,7 @@ impi_intel_linux_64_db : setup $(obj_mpi)
 impi_intel_linux_64_dv : FFLAGS = -m64 -O1 -traceback -no-wrap-margin $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_linux_64_dv : LFLAGSMKL = $(LFLAGSMKL_INTEL)
 impi_intel_linux_64_dv : LFLAGS = -static-intel
-impi_intel_linux_64_dv : FCOMPL = mpiifort
+impi_intel_linux_64_dv : FCOMPL = $(MPI_I_IFORT)
 impi_intel_linux_64_dv : FOPENMPFLAGS = -qopenmp -qopenmp-link static -liomp5
 impi_intel_linux_64_dv : obj = fds_impi_intel_linux_64_dv
 impi_intel_linux_64_dv : setup $(obj_mpi)
@@ -231,7 +242,7 @@ mpi_intel_linux_64_db : setup $(obj_mpi)
 impi_intel_linux_64_advise : FFLAGS = -ladvisor -g -m64 -O2 -ldl -traceback -no-wrap-margin -I/opt/intel19/advisor_2019/include/intel64 -L/opt/intel19/advisor_2019/lib64 $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_linux_64_advise : LFLAGSMKL = $(LFLAGSMKL_INTEL)
 impi_intel_linux_64_advise : LFLAGS = -shared-intel
-impi_intel_linux_64_advise : FCOMPL = mpiifort
+impi_intel_linux_64_advise : FCOMPL = $(MPI_I_IFORT)
 impi_intel_linux_64_advise : FOPENMPFLAGS = -qopenmp -qopenmp-link static -liomp5
 impi_intel_linux_64_advise : obj = fds_impi_intel_linux_64_advise
 impi_intel_linux_64_advise : setup $(obj_mpi)
@@ -240,7 +251,7 @@ impi_intel_linux_64_advise : setup $(obj_mpi)
 impi_intel_linux_64_vtune : FFLAGS = -g -m64 -O2 -debug inline-debug-info -D TBB_USE_THREADING_TOOLS -parallel-source-info=2 $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_linux_64_vtune : LFLAGSMKL = $(LFLAGSMKL_INTEL)
 impi_intel_linux_64_vtune : LFLAGS = -shared-intel
-impi_intel_linux_64_vtune : FCOMPL = mpiifort
+impi_intel_linux_64_vtune : FCOMPL = $(MPI_I_IFORT)
 impi_intel_linux_64_vtune : FOPENMPFLAGS = -qopenmp -qopenmp-link dynamic -liomp5
 impi_intel_linux_64_vtune : obj = fds_impi_intel_linux_64_vtune
 impi_intel_linux_64_vtune : setup $(obj_mpi)
@@ -249,7 +260,7 @@ impi_intel_linux_64_vtune : setup $(obj_mpi)
 impi_intel_linux_64_inspect : FFLAGS = -tcollect -m64 -mt_mpi -check none -warn all -O0 -auto -WB -traceback -g -fpe0 -fltconsistency -stand:f08 -no-wrap-margin $(GITINFO) $(INTELMPI_COMPINFO) $(FFLAGSMKL_INTEL)
 impi_intel_linux_64_inspect : LFLAGSMKL = $(LFLAGSMKL_INTEL)
 impi_intel_linux_64_inspect : LFLAGS = -shared-intel
-impi_intel_linux_64_inspect : FCOMPL = mpiifort
+impi_intel_linux_64_inspect : FCOMPL = $(MPI_I_IFORT)
 impi_intel_linux_64_inspect : FOPENMPFLAGS = -qopenmp -qopenmp-link static -liomp5
 impi_intel_linux_64_inspect : obj = fds_impi_intel_linux_64_inspect
 impi_intel_linux_64_inspect : setup $(obj_mpi)
