diff --git a/FDS_Compilation/makefile b/FDS_Compilation/makefile
index 8d9abec77..0059c2903 100644
--- a/FDS_Compilation/makefile
+++ b/FDS_Compilation/makefile
@@ -11,6 +11,8 @@
 
 VPATH = ../FDS_Source
 
+GITINFO_BASE=-cpp -DGITHASH_PP=\"$(build_version)+g$(commit)$(compiler)$(mpi)$(mkl)\" -DGITDATE_PP=\""$(GIT_DATE)\""
+
 # Definition of the non-MPI or MPI object variables
 
 # note: objf_serial and objf_mpi contain only Fortran routines ie not isob.c
@@ -126,72 +128,72 @@ mpi_intel_win_64 : setup_win $(objwinf_mpi)
 
 # Intel Compiler for Linux, version 11
 
-intel_linux_32 : FFLAGS = -m32 -O3 -static -vec_report0
-intel_linux_32 : CFLAGS = -m32 -O -Dpp_noappend -vec_report0
+intel_linux_32 : FFLAGS = -m32 -O3 -static  $(GITINFO_BASE)
+intel_linux_32 : CFLAGS = -m32 -O -Dpp_noappend
 intel_linux_32 : FCOMPL = ifort
 intel_linux_32 : CCOMPL = icc
 intel_linux_32 : obj = fds5_intel_linux_32
 intel_linux_32 : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_serial)
 
-intel_linux_32_profile : FFLAGS = -m32 -O2 -static -vec_report0 -p
-intel_linux_32_profile : CFLAGS = -m32 -O2 -Dpp_noappend -vec_report0 -p
+intel_linux_32_profile : FFLAGS = -m32 -O2 -static -p
+intel_linux_32_profile : CFLAGS = -m32 -O2 -Dpp_noappend -p
 intel_linux_32_profile : FCOMPL = ifort
 intel_linux_32_profile : CCOMPL = icc
 intel_linux_32_profile : obj = fds5_intel_linux_32_profile
 intel_linux_32_profile : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_serial)
 
-intel_linux_64 : FFLAGS = -m64 -O3 -static -vec_report0
-intel_linux_64 : CFLAGS = -m64 -O -Dpp_noappend -vec_report0
+intel_linux_64 : FFLAGS = -m64 -O3 -static $(GITINFO_BASE)
+intel_linux_64 : CFLAGS = -m64 -O -Dpp_noappend
 intel_linux_64 : FCOMPL = ifort
 intel_linux_64 : CCOMPL = icc
 intel_linux_64 : obj = fds5_intel_linux_64
 intel_linux_64 : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_serial)
 
-mpi_intel_linux_32 : FFLAGS = -m32 -O3 -vec_report0
-mpi_intel_linux_32 : CFLAGS = -m32 -O -Dpp_noappend -vec_report0
+mpi_intel_linux_32 : FFLAGS = -m32 -O3 $(GITINFO_BASE)
+mpi_intel_linux_32 : CFLAGS = -m32 -O -Dpp_noappend
 mpi_intel_linux_32 : FCOMPL = mpif77
 mpi_intel_linux_32 : CCOMPL = icc
 mpi_intel_linux_32 : obj = fds5_mpi_intel_linux_32
 mpi_intel_linux_32 : setup $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_mpi)
 
-mpi_intel_linux_64 : FFLAGS = -m64 -O3 -vec_report0
-mpi_intel_linux_64 : CFLAGS = -m64 -O -Dpp_noappend -vec_report0
+mpi_intel_linux_64 : FFLAGS = -m64 -O3 $(GITINFO_BASE)
+mpi_intel_linux_64 : CFLAGS = -m64 -O -Dpp_noappend
 mpi_intel_linux_64 : FCOMPL = mpif77
 mpi_intel_linux_64 : CCOMPL = icc
 mpi_intel_linux_64 : obj = fds5_mpi_intel_linux_64
 mpi_intel_linux_64 : setup $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_mpi)
 
-intel_linux_32_db : FFLAGS = -m32 -check all -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency -static -vec_report0 -ftrapuv -debug full
-intel_linux_32_db : CFLAGS = -m32 -Dpp_noappend -vec_report0
+intel_linux_32_db : FFLAGS = -m32 -check all -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency -static -ftrapuv -debug full
+intel_linux_32_db : CFLAGS = -m32 -Dpp_noappend
 intel_linux_32_db : FCOMPL = ifort
 intel_linux_32_db : CCOMPL = icc
 intel_linux_32_db : obj = fds5_intel_linux_32_db
 intel_linux_32_db : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_serial)
 
-intel_linux_64_db : FFLAGS = -m64 -check -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency -static -vec_report0
-intel_linux_64_db : CFLAGS = -m64 -Dpp_noappend -vec_report0
+intel_linux_64_db : FFLAGS = -m64 -check -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency -static
+intel_linux_64_db : CFLAGS = -m64 -Dpp_noappend
 intel_linux_64_db : FCOMPL = ifort
 intel_linux_64_db : CCOMPL = icc
 intel_linux_64_db : obj = fds5_intel_linux_64_db
 intel_linux_64_db : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_serial)
 
-mpi_intel_linux_32_db : FFLAGS = -m32 -check -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency  -vec_report0
-mpi_intel_linux_32_db : CFLAGS = -m32 -Dpp_noappend -vec_report0
+mpi_intel_linux_32_db : FFLAGS = -m32 -check -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency 
+mpi_intel_linux_32_db : CFLAGS = -m32 -Dpp_noappend
 mpi_intel_linux_32_db : FCOMPL = mpif77
 mpi_intel_linux_32_db : CCOMPL = icc
 mpi_intel_linux_32_db : obj = fds5_mpi_intel_linux_32_db
 mpi_intel_linux_32_db : setup $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_mpi)
 
-mpi_intel_linux_64_db : FFLAGS = -m64 -check -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency -vec_report0
-mpi_intel_linux_64_db : CFLAGS = -m64 -O -Dpp_noappend -vec_report0
+mpi_intel_linux_64_db : FFLAGS = -m64 -check -warn unused -O0 -auto -WB -traceback -g -fpe0 -fltconsistency
+mpi_intel_linux_64_db : CFLAGS = -m64 -O -Dpp_noappend
 mpi_intel_linux_64_db : FCOMPL = mpif77
 mpi_intel_linux_64_db : CCOMPL = icc
 mpi_intel_linux_64_db : obj = fds5_mpi_intel_linux_64_db
@@ -199,31 +201,31 @@ mpi_intel_linux_64_db : setup $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_mpi)
 
 # 32 bit
-intel_osx_32 : FFLAGS = -O3  -m32 -axSSSE3 -heap-arrays -static-intel -vec_report0
-intel_osx_32 : CFLAGS = -O3 -m32 -Dpp_noappend -Dpp_OSX -vec_report0
+intel_osx_32 : FFLAGS = -O3  -m32 -axSSSE3 -heap-arrays -static-intel
+intel_osx_32 : CFLAGS = -O3 -m32 -Dpp_noappend -Dpp_OSX
 intel_osx_32 : FCOMPL = ifort
 intel_osx_32 : CCOMPL = icc
 intel_osx_32 : obj = fds5_intel_osx_32
 intel_osx_32 : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -mmacosx-version-min=10.4 -o $(obj) $(obj_serial)
 
-mpi_intel_osx_32 : FFLAGS  = -O3 -m32 -axSSSE3 -heap-arrays -static-intel -L/opt/openmpi32/lib -vec_report0
-mpi_intel_osx_32 : CFLAGS  = -O3 -m32 -Dpp_noappend -Dpp_OSX -vec_report0
+mpi_intel_osx_32 : FFLAGS  = -O3 -m32 -axSSSE3 -heap-arrays -static-intel -L/opt/openmpi32/lib
+mpi_intel_osx_32 : CFLAGS  = -O3 -m32 -Dpp_noappend -Dpp_OSX
 mpi_intel_osx_32 : FCOMPL  = /opt/openmpi32/bin/mpif77
 mpi_intel_osx_32 : CCOMPL  = icc
 mpi_intel_osx_32 : obj = fds5_mpi_intel_osx_32
 mpi_intel_osx_32 : setup $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) -mmacosx-version-min=10.4 -o $(obj) $(obj_mpi)
 
-intel_osx_32_db : FFLAGS = -check -warn unused -O0 -m32 -g -traceback -fpe:0 -fltconsistency -FR -WB -vec_report0
-intel_osx_32_db : CFLAGS = -O0 -m32 -g -traceback -Dpp_noappend -Dpp_OSX -vec_report0
+intel_osx_32_db : FFLAGS = -check -warn unused -O0 -m32 -g -traceback -fpe:0 -fltconsistency -FR -WB
+intel_osx_32_db : CFLAGS = -O0 -m32 -g -traceback -Dpp_noappend -Dpp_OSX
 intel_osx_32_db : FCOMPL = ifort
 intel_osx_32_db : CCOMPL = icc
 intel_osx_32_db : obj = fds5_intel_osx_32_db
 intel_osx_32_db : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -mmacosx-version-min=10.4 -o $(obj) $(obj_serial)
 
-intel_osx_mpi_32_db : FFLAGS  = -check -warn unused -O0 -m32 -g -traceback -fpe:0 -vec_report0
+intel_osx_mpi_32_db : FFLAGS  = -check -warn unused -O0 -m32 -g -traceback -fpe:0
 intel_osx_mpi_32_db : CFLAGS  = -O0 -m32 -g -traceback -Dpp_noappend -Dpp_OSX
 intel_osx_mpi_32_db : FCOMPL  = /opt/openmpi32/bin/mpif77
 intel_osx_mpi_32_db : CCOMPL  = icc
@@ -233,31 +235,31 @@ intel_osx_mpi_32_db : setup $(obj_mpi)
 
 # 64 bit
 intel_osx_64 : FFLAGS = -O3 -m64 -heap-arrays -axSSSE3 -static-intel
-intel_osx_64 : CFLAGS = -O3 -m64 -Dpp_noappend -Dpp_OSX -vec_report0
+intel_osx_64 : CFLAGS = -O3 -m64 -Dpp_noappend -Dpp_OSX
 intel_osx_64 : FCOMPL = ifort
 intel_osx_64 : CCOMPL = icc
 intel_osx_64 : obj = fds5_intel_osx_64
 intel_osx_64 : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -mmacosx-version-min=10.4 -o $(obj) $(obj_serial)
 
-mpi_intel_osx_64 : FFLAGS  = -O3 -m64 -heap-arrays -axSSSE3 -static-intel -L/opt/openmpi64/lib -vec_report0
-mpi_intel_osx_64 : CFLAGS  = -O3 -m64 -Dpp_noappend -Dpp_OSX -vec_report0
+mpi_intel_osx_64 : FFLAGS  = -O3 -m64 -heap-arrays -axSSSE3 -static-intel -L/opt/openmpi64/lib
+mpi_intel_osx_64 : CFLAGS  = -O3 -m64 -Dpp_noappend -Dpp_OSX
 mpi_intel_osx_64 : FCOMPL  = /opt/openmpi64/bin/mpif77
 mpi_intel_osx_64 : CCOMPL  = icc
 mpi_intel_osx_64 : obj = fds5_mpi_intel_osx_64
 mpi_intel_osx_64 : setup $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) -mmacosx-version-min=10.4 -o $(obj) $(obj_mpi)
 
-intel_osx_64_db : FFLAGS = -check -warn unused -O0 -m64 -g -traceback -save-temps -fpe:0 -fltconsistency -fp-stack-check -FR -WB -vec_report0
-intel_osx_64_db : CFLAGS = -O0 -g -traceback -m64 -Dpp_noappend -Dpp_OSX -vec_report0
+intel_osx_64_db : FFLAGS = -check -warn unused -O0 -m64 -g -traceback -save-temps -fpe:0 -fltconsistency -fp-stack-check -FR -WB
+intel_osx_64_db : CFLAGS = -O0 -g -traceback -m64 -Dpp_noappend -Dpp_OSX
 intel_osx_64_db : FCOMPL = ifort
 intel_osx_64_db : CCOMPL = icc
 intel_osx_64_db : obj = fds5_intel_osx_64_db
 intel_osx_64_db : setup $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -mmacosx-version-min=10.4 -o $(obj) $(obj_serial)
 
-intel_osx_mpi_64_db : FFLAGS  = -check -warn unused -O0 -g -traceback -m64 -fpe:0 -fp-stack-check -save-temps -vec_report0
-intel_osx_mpi_64_db : CFLAGS  = -O0 -g -traceback -m64 -Dpp_noappend -Dpp_OSX -vec_report0
+intel_osx_mpi_64_db : FFLAGS  = -check -warn unused -O0 -g -traceback -m64 -fpe:0 -fp-stack-check -save-temps
+intel_osx_mpi_64_db : CFLAGS  = -O0 -g -traceback -m64 -Dpp_noappend -Dpp_OSX
 intel_osx_mpi_64_db : FCOMPL  = /opt/openmpi64/bin/mpif77
 intel_osx_mpi_64_db : CCOMPL  = icc
 intel_osx_mpi_64_db : obj = fds5_mpi_intel_osx_64_db
@@ -402,7 +404,7 @@ gnu_win_32 : obj     = fds5_gnu_win_32
 gnu_win_32 : $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_serial)
 
-gnu_linux_32 : FFLAGS  = -O1 -ffree-line-length-none -ffree-form
+gnu_linux_32 : FFLAGS  = -O1 -ffree-line-length-none -ffree-form $(GITINFO_BASE)
 gnu_linux_32 : CFLAGS  = -O1
 gnu_linux_32 : FCOMPL  = gfortran
 gnu_linux_32 : CCOMPL  = gcc
@@ -410,7 +412,7 @@ gnu_linux_32 : obj     = fds5_gnu_linux_32
 gnu_linux_32 : $(obj_serial)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_serial)
 
-gnu_mpi_32 : FFLAGS  = -O1
+gnu_mpi_32 : FFLAGS  = -O1 $(GITINFO_BASE)
 gnu_mpi_32 : CFLAGS  = -O1 
 gnu_mpi_32 : FCOMPL  = mpif90
 gnu_mpi_32 : CCOMPL  = mpicc
@@ -418,7 +420,7 @@ gnu_mpi_32 : obj = fds5_gnu_mpi_32
 gnu_mpi_32 : $(obj_mpi)
 	$(FCOMPL) $(FFLAGS) -o $(obj) $(obj_mpi)
 
-gnu_linux_64 : FFLAGS = -O3 -unroll -static -frecord-marker=4
+gnu_linux_64 : FFLAGS = -O3 -unroll -static -frecord-marker=4 $(GITINFO_BASE)
 gnu_linux_64 : CFLAGS = -O3 -unroll -static
 gnu_linux_64 : FCOMPL = gfortran
 gnu_linux_64 : CCOMPL = gcc
@@ -433,7 +435,7 @@ xxxmpi_gnu_linux_64 : CCOMPL = mpicc
 xxxmpi_gnu_linux_64 : obj = fds5_mpi_gnu_linux_64 
 xxxmpi_gnu_linux_64 : setup $(obj_mpi) 
 
-mpi_gnu_linux_64 : FFLAGS = -m64 -O2 -unroll
+mpi_gnu_linux_64 : FFLAGS = -m64 -O2 -unroll $(GITINFO_BASE)
 mpi_gnu_linux_64 : CFLAGS = -m64 -O2 -unroll
 mpi_gnu_linux_64 : FCOMPL = mpif77
 mpi_gnu_linux_64 : CCOMPL = mpicc 
diff --git a/FDS_Source/cons.f90 b/FDS_Source/cons.f90
index 51899fb0a..35110192d 100644
--- a/FDS_Source/cons.f90
+++ b/FDS_Source/cons.f90
@@ -80,7 +80,7 @@ CHARACTER(256) :: TERRAIN_IMAGE
 
 ! Dates, version numbers, revision numbers
 
-CHARACTER(20) :: COMPILE_DATE='null',VERSION_STRING='null'
+CHARACTER(256) :: COMPILE_DATE='null',VERSION_STRING='null'
 REAL(FB) :: VERSION_NUMBER=5.0
 INTEGER :: SVN_REVISION_NUMBER=0
  
diff --git a/FDS_Source/main.f90 b/FDS_Source/main.f90
index e045187e4..81fd7c7dc 100644
--- a/FDS_Source/main.f90
+++ b/FDS_Source/main.f90
@@ -1,3 +1,6 @@
+#ifndef GITHASH_PP
+#define GITHASH_PP "unknown"
+#endif
 PROGRAM FDS  
 
 ! Fire Dynamics Simulator, Main Program, Multiple CPU version.
@@ -97,7 +100,7 @@ WALL_CLOCK_START = WALL_CLOCK_TIME()
  
 ! Assign a compilation date (All Nodes)
 
-WRITE(VERSION_STRING,'(A)') '5.5.3'
+WRITE(VERSION_STRING,'(A)') TRIM(GITHASH_PP)
 
 IF (INDEX(mainrev,':',BACK=.TRUE.)>0) THEN
    WRITE(REVISION_DATE,'(A)',IOSTAT=IOS,ERR=5) mainrev(INDEX(mainrev,':')+1:LEN_TRIM(mainrev)-2)
