diff --git a/Build/makefile b/Build/makefile
index 483d9fc11c..eee124e39e 100644
--- a/Build/makefile
+++ b/Build/makefile
@@ -31,9 +31,9 @@ else
   else
     GIT_STAT := -dirty
   endif
-  GIT_DATE := $(shell git log -1 --format=%cd)
+  GIT_DATE := $(shell date "+%b %d, %Y  %T" --date='@${SOURCE_DATE_EPOCH}')
   GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
-  BUILD_DATE := $(shell date "+%b %d, %Y  %T")
+  BUILD_DATE := $(shell date "+%b %d, %Y  %T" --date='@${SOURCE_DATE_EPOCH}')
 # The preprocessor used by xlf uses comma as separation between preprocessing commands.
 # Commas needs to be escaped if they are to be used as part of a value in a preprocessing variable.
   BUILD_DATE_XLF := $(shell date "+%b %d\, %Y  %T")
@@ -44,7 +44,7 @@ endif
 HYPRE_INFO=-DHYPRE_PP=\"$(HYPRE_VERSION)\"
 SUNDIALS_INFO=-DSUNDIALS_PP=\"$(SUNDIALS_VERSION)\"
 
-GITINFO_BASE=-DGITHASH_PP=\"$(GIT_HASH)$(GIT_STAT)-$(GIT_BRANCH)\" -DGITDATE_PP=\""$(GIT_DATE)\""
+GITINFO_BASE=-DGITHASH_PP=\"$(build_version)+g$(commit)$(compiler)$(mpi)$(mkl)\" -DGITDATE_PP=\""$(GIT_DATE)\""
 GITINFO=-fpp $(GITINFO_BASE) -DBUILDDATE_PP=\""$(BUILD_DATE)\""
 GITINFOGNU=-cpp $(GITINFO_BASE) -DBUILDDATE_PP=\""$(BUILD_DATE)\""
 GITINFOXLF=-qsuffix=cpp=f90 '-WF,-DGITHASH_PP=\"$(GIT_HASH)\",-DGITDATE_PP=\"$(GIT_DATE)\",-DBUILDDATE_PP=\"$(BUILD_DATE_XLF)\"'
