diff --git a/FDS_Source/main.f90 b/FDS_Source/main.f90
index 4b7a8a90c..2e1106989 100644
--- a/FDS_Source/main.f90
+++ b/FDS_Source/main.f90
@@ -105,7 +105,7 @@ WALL_CLOCK_START = WALL_CLOCK_TIME()
  
 ! Assign a compilation date (All Nodes)
 
-WRITE(VERSION_STRING,'(A)') 'FDS 6.1.2'
+WRITE(VERSION_STRING,'(A)') '6.1.2+g689afcd4c'
 
 IF (INDEX(mainrev,':',BACK=.TRUE.)>0) THEN
    WRITE(REVISION_DATE,'(A)',IOSTAT=IOS,ERR=5) mainrev(INDEX(mainrev,':')+2:LEN_TRIM(mainrev)-2)
